cmd/podman/cp.go

```diff
@@ -0,0 +1,366 @@
+package main
+
+import (
+	"archive/tar"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/containers/libpod/cmd/podman/libpodruntime"
+	"github.com/containers/libpod/libpod"
+	"github.com/containers/libpod/pkg/chrootuser"
+	"github.com/containers/libpod/pkg/rootless"
+	"github.com/containers/storage"
+	"github.com/containers/storage/pkg/archive"
+	"github.com/containers/storage/pkg/chrootarchive"
+	"github.com/containers/storage/pkg/idtools"
+	digest "github.com/opencontainers/go-digest"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+var (
+	cpDescription = "Copy files/folders between a container and the local filesystem"
+	cpCommand     = cli.Command{
+		Name:         "cp",
+		Usage:        "Copy files/folders between a container and the local filesystem",
+		Description:  cpDescription,
+		Action:       cpCmd,
+		ArgsUsage:    "CONTAINER:SRC_PATH DEST_PATH | SRC_PATH|- CONTAINER:DEST_PAT",
+		OnUsageError: usageErrorHandler,
+	}
+)
+
+func cpCmd(c *cli.Context) error {
+	args := c.Args()
+	if len(args) != 2 {
+		return errors.Errorf("you must provide source path and destination path")
+	}
+	srcPath := args[0]
+	destPath := args[1]
+
+	if isContainerPathFormat(srcPath) && isContainerPathFormat(destPath) {
+		return errors.Errorf("please provide only one container working dorectory")
+	}
+	if !isContainerPathFormat(srcPath) && !isContainerPathFormat(destPath) {
+		return errors.Errorf("you must provade one container ID")
+	}
+
+	if os.Geteuid() != 0 {
+		rootless.SetSkipStorageSetup(true)
+	}
+
+	runtime, err := libpodruntime.GetRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "could not get runtime")
+	}
+	defer runtime.Shutdown(false)
+
+	if os.Geteuid() != 0 {
+		if driver := runtime.GetConfig().StorageConfig.GraphDriverName; driver != "vfs" {
+			// Do not allow to mount a graphdriver that is not vfs if we are creating the userns as part
+			// of the mount command.
+			return fmt.Errorf("cannot mount using driver %s in rootless mode", driver)
+		}
+
+		became, ret, err := rootless.BecomeRootInUserNS()
+		if err != nil {
+			return err
+		}
+		if became {
+			os.Exit(ret)
+		}
+	}
+
+	if isContainerPathFormat(destPath) {
+		name := strings.Split(destPath, ":")[0]
```

Comment:
Should only do the strings.Split onece and set them to src and dest.  Probably better to do a Splitn2() and then verify that their are two and objects.

Reply:
```go
destPathArr, destIsValidPath := isContainerPathFormat(destPath)
if destIsValidPath {
	name := destPathArr[0]
	destination := destPathArr[1]
        ...
}
```  
 isContainerPathFormat does SplitN and returns the slice and bool 
```go
func isContainerPathFormat(path string) ([]string, bool) {
	pathArr := strings.SplitN(path, ":", 2)
	if len(pathArr) == 2 {
		return pathArr, true
	}
	return nil, false
}
```